### DISCUSSION TITLE
Timer now throws an error in undefined case

### DISCUSSION THREAD

**Discussion by FraserLee** [bevyengine/bevy#3524] [General]
### Minimum Sample
```rust
use bevy::prelude::Timer;
use bevy::utils::Duration;

fn main() {
    let mut timer = Timer::from_seconds(0.0, true);
    timer.tick(Duration::from_millis(1000));
}
```

In 0.5 this will internally cause a `nan f32`, which will saturate to `0u32` and not error. As of #2362 it will throw an error and crash the program, since this piece of math has been converted to `f128`, which doesn't saturate on casts.

I'm a bit unsure whether to put in a PR to fix it, since the usecase here is fairly marginal (eg setting a timer to 0 duration to ensure it will hit on the next loop, etc). A fix here, on the other hand, could possibly have some performance penalties.

I'm fairly new to bevy, what's the move?

**Comment by DJMcNab** [comment]
A timer with 0 time is malformed. We should instead panic on new if the 0 `Duration` is added.

The use case of wanting the Timer to 'trigger next time' seems like it should just be stored in a piece of non-`Timer` state.

**Reply by mockersf** [reply]
Bevy already panic for a negative duration, it should panic for `0.0` too 👍

**Reply by DJMcNab** [reply]
As far as I know, Duration is by definition non-negative?

How do we panic on a negative duration?

**Reply by mockersf** [reply]
I mean, when calling `from_seconds` with a `f32`, Bevy creates a std `Duration` that panics for a negative value

**Comment by mockersf** [comment]
To have a `Timer` that is triggered at creation, I usually go with something like
```rust
    let duration = Duration::from_secs_f32(19.0);
    let mut new_timer = Timer::new(duration, true);
    new_timer.set_elapsed(duration);
```
